src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,173 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head = null;
+    private Node<T> tail = null;
+    private int size;
+
+    public Node<T> getHead() {
+        return head;
+    }
+
+    public Node<T> getTail() {
+        return tail;
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkForValidIndex(index);
+        Node<T> newNode = new Node<>(value,null, null, size);
+        if (index < size) {
+            connectToNext(newNode, getNodeByIndex(index));
+        }
+        if (index > 0) {
+            connectToPrev(newNode, getNodeByIndex(index - 1));
```

Comment:
I think I understand what you did there, with assigning the nodes based on the index instead of strict cases like if the list is empty, or `index == size`, or index is the middle section. It is quite original, and I do not see any problems with that so far. so I think if it works we can stick with that. 
The only thing I wanted to mention is to not call `getNodeByIndex()` twice, have a var, and get a second one from the first one